heroku/cli
# refactor(redis-v5): Moving command 'redis:credentials' to CLI

## Description

Moves the `redis:credentials` command to CLI, converting to OClif Core and TS.

There's a small refactor on the shared lib file to add the last required function plus improve the type definition for the responses from Shogun API.

## Test instructions

You can test by pulling this branch, installing dependencies with `yarn` and building with `yarn build`. Then run the following commands, knowing that in an app without any Redis add-on, you should see an error and in apps with more than one Redis add-on you should see an error unless you specify the DATABASE arg.

```
./bin/run redis:credentials --help                                  # always test that help looks ok.
./bin/run redis:credentials -a test_app_without_redis               # shows error.
./bin/run redis:credentials -a test_app_single_redis                # shows credentials.
./bin/run redis:credentials -a test_app_single_redis --reset        # resets database credentials.
./bin/run redis:credentials -a test_app_multiple_redis              # shows error.
./bin/run redis:credentials addon-name -a test_app_multiple_redis   # shows credentials for selected add-on.
```

You can add Heroku Redis add-ons to any test app with `heroku addons:create heroku-redis -a test-app`.

SOC2 Compliance
[GUS Work Item](https://gus.lightning.force.com/a07EE00001axS4gYAE)
